python/cudf/cudf/io/parquet.py

```diff
@@ -3,12 +3,106 @@
 import warnings
 
 import pyarrow.parquet as pq
+from pyarrow.compat import guid
 
 import cudf
 import cudf._libxx.parquet as libparquet
 from cudf.utils import ioutils
 
 
+def _get_partition_groups(df, partition_cols, preserve_index=False):
+    df = df.sort_values(partition_cols)
+    if not preserve_index:
+        df = df.reset_index(drop=True)
+    divisions = df[partition_cols].drop_duplicates()
+    splits = df[partition_cols].searchsorted(divisions, side="left")
+    splits = splits.tolist() + [len(df[partition_cols])]
+    return [
+        df.iloc[splits[i] : splits[i + 1]].copy(deep=False)
+        for i in range(0, len(splits) - 1)
+    ]
+
+
+def _mkdir_if_not_exists(fs, path):
+    if fs._isfilestore() and not fs.exists(path):
+        try:
+            fs.mkdir(path)
```

Comment:
Can this all be simplified to `fs.mkdirs(path, exist_ok=True)`  ?

Reply:
Hmm - This doesn't seem to work for `LocalFileSystem` in the tests

Reply:
Is this resolved?